src/sage/categories/modules_with_basis.py

```diff
@@ -2065,8 +2065,18 @@ def map_coefficients(self, f):
                 sage: a = s([2,1]) + 2*s([3,2])                                         # needs sage.combinat sage.modules
                 sage: a.map_coefficients(lambda x: x * 2)                               # needs sage.combinat sage.modules
                 2*s[2, 1] + 4*s[3, 2]
+
+            TESTS:
+
+            map_coefficients works on Matrix::
+
+                sage: M = Matrix([[1, 0], [0, 1]])
+                sage: f = lambda x: x + 1
+                sage: M.map_coefficients(f)             # This actually failed at some point!!! See #37320
```

Comment:
```suggestion
                sage: M.map_coefficients(f)
```
You already said this basically. No need to sensationalize either.